Gabriella/Change phone to date

nuovonatura opened the pull request:
Fix #121 on behalf of Gabriella

GabriellaTeh committed:
change-phone-to-date

GabriellaTeh committed:
change date to dateTime

dearvae committed:
Merge pull request #1 from dearvae/beining/addCommand

Beining/add command

GabriellaTeh committed:
Update JsonAdaptedTask.java

GabriellaTeh committed:
fix variable names

GabriellaTeh committed:
fix variable names

dearvae committed:
merge with master

dearvae committed:
merge with head

dearvae committed:
resolve conflict

nuovonatura committed:
Merge pull request #131 from dearvae/beining/help

Beining/help

nuovonatura committed:
Adapt StringUtil to new Title field criteria

Updated the containsWordIgnoreCase method in StringUtil to allow
multiple words as keyword.

Since in PlaNus, the title attribute of a task is no longer required to
be a single word or a String without space compared to its previous
field in AddressBook3, the constrain for no multiple words as keyword
for checking title is removed.

Test cases adapted.

GabriellaTeh commented on src/main/java/seedu/address/logic/parser/ParserUtil.java:
```diff
@@ -51,18 +51,18 @@ public static Title parseTitle(String title) throws ParseException {
      * Leading and trailing whitespaces will be trimmed.
      *
-     * @throws ParseException if the given {@code phone} is invalid.
+     * @throws ParseException if the given {@code date} is invalid.
```
this should be dateTime

GabriellaTeh commented on src/main/java/seedu/address/model/task/DateTime.java:
```diff
@@ -0,0 +1,75 @@
+    }
+
+    /**
+     * Constructs an empty Phone when user didn't provide the phone field.
```
Should be "constructs an empty DateTime when user does not provide the dateTime field

GabriellaTeh commented on src/test/java/seedu/address/logic/parser/AddCommandParserTest.java:
```diff
@@ -48,47 +48,47 @@ public void parse_allFieldsPresent_success() {
 
         // missing phone field
-        expectedTask = new TaskBuilder(AMY).withTags().withDefaultPhone().build();
+        expectedTask = new TaskBuilder(AMY).withTags().withDefaultDateTime().build();
```
comment should be dateTime

GabriellaTeh commented on src/test/java/seedu/address/logic/parser/EditCommandParserTest.java:
```diff
@@ -82,17 +82,17 @@ public void parse_invalidPreamble_failure() {
     public void parse_invalidValue_failure() {
         assertParseFailure(parser, "1" + INVALID_TITLE_DESC, Title.MESSAGE_CONSTRAINTS); // invalid title
-        assertParseFailure(parser, "1" + INVALID_PHONE_DESC, Phone.MESSAGE_CONSTRAINTS); // invalid phone
+        assertParseFailure(parser, "1" + INVALID_DATE_TIME_DESC, DateTime.MESSAGE_CONSTRAINTS); // invalid phone
```
Comment should be invalid dateTime

GabriellaTeh commented on src/test/java/seedu/address/logic/parser/EditCommandParserTest.java:
```diff
@@ -82,17 +82,17 @@ public void parse_invalidPreamble_failure() {
 
         // invalid phone followed by valid description
-        assertParseFailure(parser, "1" + INVALID_PHONE_DESC + DESCRIPTION_DESC_AMY, Phone.MESSAGE_CONSTRAINTS);
+        assertParseFailure(parser, "1" + INVALID_DATE_TIME_DESC + DESCRIPTION_DESC_AMY, DateTime.MESSAGE_CONSTRAINTS);
```
this comment also

GabriellaTeh commented on src/test/java/seedu/address/testutil/TaskBuilder.java:
```diff
@@ -17,12 +17,12 @@
 
     public static final String DEFAULT_TITLE = "School work";
-    public static final String DEFAULT_PHONE = "85355255";
+    public static final String DEFAULT_DATE = "01-01-2020 12:00";
```
should DEFAULT_DATE_TIME

GabriellaTeh commented on src/test/java/seedu/address/testutil/TaskBuilder.java:
```diff
@@ -74,19 +74,19 @@ public TaskBuilder withAddress(String address) {
     /**
-     * Sets the {@code Phone} of the {@code Task} that we are building to be default phone.
-     * Simulates the situation that the task is created without a phone field.
+     * Sets the {@code Phone} of the {@code Task} that we are building to be default DateTime.
```
should be dateTime instead of Phone

GabriellaTeh committed:
Update ParserUtil.java

GabriellaTeh committed:
Update DateTime.java

GabriellaTeh committed:
Update AddCommandParserTest.java

GabriellaTeh committed:
Update EditCommandParserTest.java

GabriellaTeh committed:
Update TaskBuilder.java

GabriellaTeh committed:
Merge branch 'master' into change-phone-to-date

GabriellaTeh committed:
Update EditCommand.java

GabriellaTeh committed:
Update EditCommandParser.java

GabriellaTeh committed:
Update AddCommandParserTest.java

GabriellaTeh committed:
Update EditCommand.java

nuovonatura committed:
Remove extra Address referenced and fix test cases

Remove address class calls from the previous merge as it is replaced
with type class already.

Fix test cases in EditCommandParserTest as the method called on adding
type to EditTaskDescriptor is wrong.

MarcTzh commented on src/main/java/seedu/address/logic/commands/AddCommand.java:
```diff
@@ -21,13 +21,13 @@
             + "Parameters: "
             + PREFIX_TITLE + "TITLE "
-            + PREFIX_PHONE + "PHONE "
+            + PREFIX_DATE_TIME + "DATE "
```
Should be DATE_TIME

MarcTzh commented on src/main/java/seedu/address/logic/parser/CliSyntax.java:
```diff
@@ -7,7 +7,7 @@
     /* Prefix definitions */
     public static final Prefix PREFIX_TITLE = new Prefix("title:");
-    public static final Prefix PREFIX_PHONE = new Prefix("p/");
+    public static final Prefix PREFIX_DATE_TIME = new Prefix("date:");
```
does this prefix make sense or do we need to revise it since its date + time now, maybe it should be `due:` for example

MarcTzh commented on src/main/java/seedu/address/model/task/DateTime.java:
```diff
@@ -0,0 +1,75 @@
+    public static final String VALIDATION_REGEX =
+            "^(3[01]|[12][0-9]|0[1-9])-(1[0-2]|0[1-9])-[0-9]{4} (2[0-3]|[01]?[0-9]):([0-5]?[0-9])$";
+
+    private static final String DEFAULT_DATE_TIME = "";
```
What is the purpose of this design wise? Is this to act as a null value without throwing NPE?

MarcTzh commented on src/main/java/seedu/address/logic/parser/ParserUtil.java:
```diff
@@ -51,18 +51,18 @@ public static Title parseTitle(String title) throws ParseException {
-        String trimmedPhone = phone.trim();
-        if (!Phone.isValidPhone(trimmedPhone)) {
-            throw new ParseException(Phone.MESSAGE_CONSTRAINTS);
+    public static DateTime parseDateTime(String dateTime) throws ParseException {
```
The rationale behind using java String#trim on `Phone` was to remove leading and trailing whitespaces. When considering the user's `dateTime` input do we need to consider white spaces inside the string as well? For e.g. "01-01-2020 1200" vs "01-01-2020   1200" if we are not going to trim it then perhaps we need to consider adding an exception for this and add the corresponding unit test.

MarcTzh commented on src/test/java/seedu/address/testutil/EditTaskDescriptorBuilder.java:
```diff
@@ -84,4 +84,6 @@ public EditTaskDescriptorBuilder withTags(String... tags) {
     public EditTaskDescriptor build() {
         return descriptor;
     }
+
```
Remove this

MarcTzh commented on src/test/java/seedu/address/testutil/TaskBuilder.java:
```diff
@@ -74,19 +74,19 @@ public TaskBuilder withType(String type) {
 
     /**
-     * Sets the {@code Phone} of the {@code Task} that we are building.
+     * Sets the {@code date} of the {@code Task} that we are building.
```
rename this

MarcTzh commented on src/test/java/seedu/address/testutil/TaskUtil.java:
```diff
@@ -46,7 +46,7 @@ public static String getTaskDetails(Task task) {
         StringBuilder sb = new StringBuilder();
         descriptor.getTitle().ifPresent(title -> sb.append(PREFIX_TITLE).append(title.title).append(" "));
-        descriptor.getPhone().ifPresent(phone -> sb.append(PREFIX_PHONE).append(phone.value).append(" "));
+        descriptor.getDateTime().ifPresent(date -> sb.append(PREFIX_DATE_TIME).append(date.value).append(" "));
```
rename the date

MarcTzh reviewed (commented):
Please make the changes

GabriellaTeh committed:
Update AddCommand.java

GabriellaTeh committed:
Update EditTaskDescriptorBuilder.java

GabriellaTeh committed:
Update TaskBuilder.java

GabriellaTeh committed:
Update TaskUtil.java

BobbyZhouZijian reviewed (approved):
LGTM
